src/select/select.tsx

```diff
@@ -145,64 +124,104 @@ class Select<T> extends React.Component<Props<T>, State<T>> {
+        });
     }
 
-    private toggleOptionsIfNotDisabled() {
-        if (!this.props.disabled) {
-            this.toggleOptionsList();
-        }
+    private onFilterChange(event: React.ChangeEvent<HTMLInputElement>) {
+        this.setState({
+            filterValue: event.target.value,
+        });
     }
 
-    private toggleOptionsList() {
-        this.setState(
-            (prevState) => {
-                return {
-                    expanded: !prevState.expanded,
-                    options: this.props.options,
-                };
-            },
-            () => {
-                this.focusSearchInput();
-            }
-        );
+    private setRootNode(node: HTMLDivElement) {
+        this.rootNode = node;
     }
 
-    private focusSearchInput() {
-        if (this.props.searchable && this.searchInputRef.current) {
-            this.searchInputRef.current.focus();
-        }
+    private renderStandardTrigger() {
+        const label = this.getLabel();
+        const disabledClass = this.props.disabled ? ' disabled' : '';
+        const placeholderClass = !label && this.props.placeholder ? ' has-placeholder' : '';
+        return (
+            <div
+                role="button"
+                data-qa={`select-toggle--standard-${this.props.id}`}
+                className={`select-option-toggle select-option-toggle--standard${disabledClass}${placeholderClass}`}
+                aria-haspopup="listbox"
+                onClick={this.toggleOptionsList}
+            >
+                <span className={`text`} data-qa={`select-toggle--standard-text-${this.props.id}`}>
+                    {label || this.props.placeholder}
+                </span>
+                <span className="select-option-toggle--icon-dropdown" />
+            </div>
+        );
     }
 
-    private isSelected(option: Option<T>) {
-        return isEqual(this.props.value, option.value);
+    private renderSearchableTrigger() {
+        const label = this.getLabel();
+        const disabledClass = this.props.disabled ? ' disabled' : '';
+        const placeholderClass = this.state.isExpanded || !label ? ' has-faded-placeholder' : '';
+        return (
+            <div
+                role="button"
+                data-qa={`select-toggle--search-${this.props.id}`}
+                className={`select-option-toggle select-option-toggle--search${disabledClass}${placeholderClass}`}
+                aria-haspopup="listbox"
+            >
+                {this.getSearchInput()}
+            </div>
+        );
     }
 
-    private filterOptions(event: React.ChangeEvent<HTMLInputElement>) {
-        const filteredOptions = this.props.options.filter((option) => matches(event.target.value, option.label));
-        this.setState({
-            options: filteredOptions,
-        });
+    private getSearchInput() {
+        return (
+            <>
+                <span className="select-option-toggle--icon-filter" />
+                <input
```

Comment:
Wasn't there a component for this?

Reply:
Yes, but this one is behaving a bit differently. So I either can extend the current one, or use this solution.

Reply:
What is the difference?

Reply:
The component for SearchInput was missing some options to fit ac fully, but It's fixed now and `SearchInput` is integrated into `select.tsx`.